tests/lib/prepare.sh

```diff
@@ -775,6 +775,39 @@ uc20_build_corrupt_kernel_snap() {
     rm -rf "$REPACKED_DIR"
 }
 
+uc_write_bootstrap_wrapper() {
+    local SKELETON_PATH="$1"
+    local INJECT_ERR="${2:-false}"
+
+    cp -a /usr/lib/snapd/snap-bootstrap "$SKELETON_PATH"/usr/lib/snapd/snap-bootstrap.real
+    cat <<'EOF' >"$SKELETON_PATH"/usr/lib/snapd/snap-bootstrap
+#!/bin/sh
+set -eux
+if [ "$1" != initramfs-mounts ]; then
+    exec /usr/lib/snapd/snap-bootstrap.real "$@"
+fi
+beforeDate="$(date --utc '+%s')"
+/usr/lib/snapd/snap-bootstrap.real "$@"
+if [ -d /run/mnt/data/system-data ]; then
+    touch /run/mnt/data/system-data/the-tool-ran
+fi
+# also copy the time for the clock-epoch to system-data, this is
+# used by a specific test but doesn't hurt anything to do this for
```

Comment:
It would be nice to find out which "specific test" needs it and add it in the comment. I really wonder if that is still needed.